modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

```diff
@@ -4616,6 +4623,13 @@ public CacheGroupDescriptor cacheGroupDescriptor(int grpId) {
      * @return Cache descriptor.
      */
     public @Nullable DynamicCacheDescriptor cacheDescriptor(int cacheId) {
+        Map<Integer, String> cacheIdToNameMapping = cacheIDToNameMapping();
+        String name = cacheIdToNameMapping.get(cacheId);
+
+        if (name != null) {
+            return cacheDescriptors().get(name);
+        }
+
         for (DynamicCacheDescriptor cacheDesc : cacheDescriptors().values()) {
```

Comment:
Why do we need to iterate descriptors? Maybe we can rely on cacheIdToNameMapping and get rid of that code?